### PR TITLE
Update rust toolchain to 1.80.1

### LINKED ISSUE #98: Update to Rustc 1.80.0
Some packages require 1.80.0, it would be nice to update the toolchain at least to it

### PR DESCRIPTION
**Pull request by apljungquist**
1.80.1 is chosen because it the latest version that has had an entire release cycle's worth of scrutiny at this point.

`Cargo.lock`:
- Update time to avoid E0282 on format_description/parse/mod.rs:83:9.

Resolves #98

### CODE REVIEW

File: crates/axevent-sys/build.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@ fn populated_bindings(dst: &path::PathBuf) {
         .parse_callbacks(Box::new(bindgen::CargoCallbacks::new()))
         .layout_tests(false);
     for path in library.include_paths {
-        bindings = bindings.clang_args(&["-F", &path.to_str().unwrap()]);
+        bindings = bindings.clang_args(&["-F", (path.to_str().unwrap())]);
```
Comment (guoxe) [inline crates/axevent-sys/build.rs:13]: Why is this changed?

Comment (apljungquist) [reply crates/axevent-sys/build.rs:13]: Clippy started complaining about it.

Sorry about the lack of explanation in the PR description; Because this project is set to create the default commit message by concatenating all commits messages I am not updating the PR description and motivations are more likely to be found in commits prefixed `s:` for _squash_.

**Review by guoxe — APPROVED** [review]
